streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -1331,6 +1307,20 @@ private void validateRackAwarenessConfiguration() {
         });
     }
 
+    private static void validateProcessingConfiguration(final String processingModeConfig) {
+        if (processingModeConfig.equals(EXACTLY_ONCE)) {
+            throw new ConfigException(String.format("Configuration parameter `%s` was removed in the 4.0.0 release. " +
+                "Please use `%s` instead. Note that this requires broker version 2.5+ so you should prepare "
+                + "to upgrade your brokers if necessary.", EXACTLY_ONCE, EXACTLY_ONCE_V2));
+        }
+        if (processingModeConfig.equals(EXACTLY_ONCE_BETA)) {
+            throw new ConfigException(String.format("Configuration parameter `%s` was removed in the 4.0.0 release. " +
+                "Please use `%s` instead. Note that this requires broker version 2.5+ so you should prepare "
```

Comment:
If users are already using BETA then their brokers should be on 2.5 already, since the semantics of BETA and V2 are exactly the same. I think we should just say "... Please use .. instead which is the new name for the same processing semantics".

Reply:
Done